### PR TITLE
Update nmit corrlation matrix

### PR DESCRIPTION
**Pull request by elong0527**
Increase effciency by using pandas rountine function.

### CODE REVIEW

File: q2_longitudinal/_utilities.py [hunk 1]
```diff
@@ -580,25 +580,20 @@ def _stats_and_visuals(output_dir, pairs, metric, group_column,
...
         temporal correlation method, "kendall", "pearson", "spearman"
-
     '''
-    results = {}
 
     # Start to calculate temporal correlation
-    for id_key in id_set:
-        table_id = table.loc[individual_id == id_key]
-        results[id_key] = table_id.corr(method=corr_method).fillna(0)
+    table["individual_id"] = individual_id
```
Comment (nbokulich) [inline q2_longitudinal/_utilities.py]: is this necessary? in `_utilities._nmit` it looks like this is still:
``individual_id = sample_md[individual_id_column]``

so could this line be removed and line 596 be?:
``results = table.groupby([individual_id_column]).corr(method=corr_method).fillna(0)``

(``individual_id_column`` would need to be added as an input parameter to this function)

Comment (elong0527) [reply q2_longitudinal/_utilities.py]: Table is actually an OTUs table. I am not sure how "individual_id_column" can be indexed by Pandas if I do not add the column in the OTUs table.

**Comment by elong0527** [comment]
Thanks @nbokulich for your patience detail instructions! The `flake8` is great. 

I am trying to use `notetests`, however I am using a Windows computer and I could not run unittest without `qimme2`. Could you help me run the unit test?

The iPython Notebook on my Unix Virtual machine does not work properly. I may need more time to figure it out.

**Comment by elong0527** [comment]
Hi @nbokulich , I added a screening functon so that we do not have to run all taxa in the tutorial to have a lightweight tutorial. 

And in the tutorial, we can use "pearson" correlation instead of "kendall". So the tutorial will run much master (Even through we suggest "kendall" in our paper).

Below is an example code for the NMIT tutorial. 
 
```
import numpy as np
import pandas as pd
from scipy import linalg

taxa = pd.read_csv("Documents/GitHub/taxa.csv") # This is ECAM taxa table in tutorial_data
meta = pd.read_csv("Documents/GitHub/meta.csv") # This is ECAM metadata table in tutorial_data
taxa_s = _temporal_screening(taxa, abs_or_rel= "rel", rel_threshold= 0.001, pct_threshold= 0.15) # First screening taxa
_nmit(taxa_s, sample_md= meta, corr_method= "pearson", dist_method= "fro", individual_id_column= "studyid") # Then perform nmit test
```

PS: I got a warning "SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead" When I run the code in Ipython Notebook. However, I am not sure how could I fix it.

**Comment by nbokulich** [comment]
Thanks @elong0527 ! I was able to pull down your changes to fix the unit tests (the changes to ``_temporal_corr`` were causing some tests to break since it now outputs a pd.DataFrame instead of a dict of dataframes)

I wound up pushing your/my changes in a separate PR ( #4 ) as I needed to pull into a separate branch for testing. So I am closing your PR but all of your changes have been approved/merged in that separate PR.

The only change that I made to your latest version of the code was to remove the filtering function — while I really like including functions like this, the qiime2 developers won't want to include that since it is redundant with functions already in qiime2.

**Pull request closed without merging**